runtime/near-vm-runner/src/cache.rs

```diff
@@ -304,7 +353,8 @@ pub mod wasmer2_cache {
             wasm_code: &[u8],
             config: &VMConfig,
             cache: Option<&dyn CompiledContractCache>,
-            store: &wasmer::Store) -> Result<wasmer::Module, VMError> = {
+            store: &wasmer::Store
```

Comment:
@olonho not sure it's correct to pass `store` into this function -- it might be the case that the cached contract mixes two stores that way. From my reading of the code, this should be fine. if it is, we probably should add an assert with [`Store::same`](https://docs.rs/wasmer/2.0.0/wasmer/struct.Store.html#method.same).

Reply:
Discussed with olonho, this behaves correctly today, so no immediate action is required. But would be nice to straighten this up some day!